Assets/Mirror/Runtime/NetworkConnectionToClient.cs

```diff
@@ -9,16 +10,124 @@ public class NetworkConnectionToClient : NetworkConnection
 
         public override string address => Transport.activeTransport.ServerGetClientAddress(connectionId);
 
-        public NetworkConnectionToClient(int networkConnectionId) : base(networkConnectionId) { }
+        // batching from server to client.
+        // fewer transport calls give us significantly better performance/scale.
+        //
+        // for a 64KB max message transport and 64 bytes/message on average, we
+        // reduce transport calls by a factor of 1000.
+        //
+        // depending on the transport, this can give 10x performance.
+        //
+        // Dictionary<channelId, batch> because we have multiple channels.
+        internal class Batch
+        {
+            // each batch needs a writer for batching
+            // => we allocate one writer per channel
+            // => it grows to Transport MaxMessageSize automatically
+            // TODO maybe use a pooled writer and return when disconnecting?
+            internal NetworkWriter writer = new NetworkWriter();
+
+            // each channel's batch has its own lastSendTime.
+            // (use NetworkTime for maximum precision over days)
+            //
+            // channel batches are full and flushed at different times. using
+            // one global time wouldn't make sense.
+            // -> we want to be able to reset a channels send time after Send()
+            //    flushed it because full. global time wouldn't allow that, so
+            //    we would often flush in Send() and then flush again in Update
+            //    even though we just flushed in Send().
+            // -> initialize with current NetworkTime so first update doesn't
+            //    calculate elapsed via 'now - 0'
+            internal double lastSendTime = NetworkTime.time;
+        }
+        Dictionary<int, Batch> batches = new Dictionary<int, Batch>();
+
+        // batch interval is 0 by default, meaning that we send immediately.
+        // (useful to run tests without waiting for intervals too)
+        float batchInterval;
+
+        public NetworkConnectionToClient(int networkConnectionId, float batchInterval)
+            : base(networkConnectionId)
+        {
+            this.batchInterval = batchInterval;
+        }
+
+        Batch GetBatchForChannelId(int channelId)
+        {
+            // get existing or create new writer for the channelId
+            Batch batch;
+            if (!batches.TryGetValue(channelId, out batch))
+            {
+                batch = new Batch();
+                batches[channelId] = batch;
+            }
+            return batch;
+        }
+
+        void SendBatch(int channelId, Batch batch)
+        {
+            // send batch
+            Transport.activeTransport.ServerSend(connectionId, channelId, batch.writer.ToArraySegment());
+
+            // clear batch
+            batch.writer.Position = 0;
```

Comment:
```suggestion
            batch.writer.Reset();
```

Length needs to be reset too

Reply:
@JesusLuvsYooh confirmed this works